navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigation.java

```diff
@@ -601,7 +601,11 @@ public void onProgressChange(Location location, RouteProgress routeProgress) {
     Timber.v("Arrived event occurred");
     sessionState = sessionState.toBuilder().arrivalTimestamp(new Date()).build();
     NavigationMetricsWrapper.arriveEvent(sessionState, routeProgress, location);
-    endNavigation();
+    // Remove all listeners except the onProgressChange by passing in null.
+    navigationEventDispatcher.removeOffRouteListener(null);
+    navigationEventDispatcher.removeMilestoneEventListener(null);
```

Comment:
Isn't this line called before the Arrival Milestone is able to be processed? With this line, my arrival milestone fails to fire and without it everything works fine. Maybe I am doing something wrong or misunderstanding some aspect of the Arrival/Destination Milestone?